extensions/pkg/controller/healthcheck/reconciler.go

```diff
@@ -275,13 +275,15 @@ type condition struct {
 }
 
 func (r *reconciler) updateExtensionConditions(ctx context.Context, extension extensionsv1alpha1.Object, conditions ...condition) error {
+	newConditions := []gardencorev1beta1.Condition{}
 	for _, cond := range conditions {
 		if c := v1beta1helper.GetCondition(extension.GetExtensionStatus().GetConditions(), gardencorev1beta1.ConditionType(cond.healthConditionType)); c != nil {
 			cond.builder.WithOldCondition(*c)
 		}
 		updatedCondition, _ := cond.builder.WithClock(clock.RealClock{}).Build()
-		extension.GetExtensionStatus().SetConditions(v1beta1helper.MergeConditions(extension.GetExtensionStatus().GetConditions(), updatedCondition))
+		newConditions = append(newConditions, updatedCondition)
```

Comment:
This removes conditions added by other parties. While this is rather a theoretical problem (I am not aware of a real-world example where multiple components contribute to the conditions), it would be nice if you could rather explicitly remove the conditions that might have been written by the health check controller in earlier versions but are no longer written now (e.g., `ShootSystemComponentsHealthy` or `ShootControlPlaneHealthy`).